packages/near-membrane-base/src/membrane.ts

```diff
@@ -1052,20 +1050,32 @@ export function createMembraneMarshall(
         ) {
             delete lazyState[key];
             let safeDesc;
-            foreignCallableGetOwnPropertyDescriptor(
-                getTransferablePointer(target),
-                key,
-                (configurable, enumerable, writable, valuePointer, getPointer, setPointer) => {
-                    safeDesc = createDescriptorFromMeta(
-                        configurable,
-                        enumerable,
-                        writable,
-                        valuePointer,
-                        getPointer,
-                        setPointer
-                    );
-                }
-            );
+            const foreignTargetPointer = getTransferablePointer(target);
+            const activity = instrumentation
+                ? instrumentation.startActivity('callableGetOwnPropertyDescriptor')
+                : undefined;
+            try {
+                foreignCallableGetOwnPropertyDescriptor(
+                    foreignTargetPointer,
+                    key,
+                    (configurable, enumerable, writable, valuePointer, getPointer, setPointer) => {
+                        safeDesc = createDescriptorFromMeta(
+                            configurable,
+                            enumerable,
+                            writable,
+                            valuePointer,
+                            getPointer,
+                            setPointer
+                        );
+                    }
+                );
+                activity?.stop();
```

Comment:
Move to `finally` block?

Reply:
Finally blocks are can affect execution flow so I try to avoid them unless we're dealing with a `return` statement.

Reply:
It only affects execution flow if there's a return in the finally so I'll sit with it a bit.